runtime/j9vm/j9scar.tdf

```diff
@@ -334,3 +334,7 @@ TraceExit=Trc_SC_GCNoCompact_Exit Overhead=1 Level=1 Template="JVM_GCNoCompact()
 TraceEntry=Trc_SC_LoadSystemLibrary_Entry NoEnv Overhead=1 Level=1 Template="JVM_LoadSystemLibrary(name=%s)"
 TraceExit=Trc_SC_LoadSystemLibrary_Exit NoEnv Overhead=1 Level=1 Template="JVM_LoadSystemLibrary -- return 0x%zx"
 TraceEvent=Trc_SC_LoadSystemLibrary_LoadFailed NoEnv Overhead=1 Level=1 Template="JVM_LoadSystemLibrary(name=%s) failed to load"
+
+TraceEntry=Trc_SC_RegisterSignal_Entry NoEnv Overhead=1 Level=1 Template="JVM_RegisterSignal(signal=%d, handler=%p)"
```

Comment:
`NoEnv` tracepoints have to attempt to lookup the current J9VMThread to store into the thread's tracebuffer.  Since every invocation of this function will need to do lookup twice (once for enter/exit), can you make these take the env?

Reply:
`JVM_RegisterSignal` doesn't get access to `JNIEnv *env`. Used `javaVM->mainThread` as `env`. Is `javaVM->mainThread` a satisfactory alternative for `env`?

Reply:
`javaVM->mainThread` won't be a satisfactory alternative for `env`. `JVM_RegisterSignal` may not always be invoked from `javaVM->mainThread`. in such scenarios, `env` will be set incorrectly. so, we have to go with the `NoEnv` approach for `JVM_RegisterSignal` tracepoints since `JVM_RegisterSignal` doesn't get access to `JNIEnv *env`. @DanHeidinga am i correct?

Reply:
You should be able to use `javaVM->internalVMFunctions->currentVMThread(javaVM)`

Reply:
updated tracepoints to take `env` as input.